core/src/test/java/io/questdb/griffin/TimestampQueryTest.java

```diff
@@ -1360,6 +1360,110 @@ public void testTimestampSymbolDateAdd() throws Exception {
+            assertSql(query, expected);
+            // test mid-case
+            expected = "seq_num\ttimestamp\n" +
+                    "3\t2022-11-21T00:00:00.000000Z\n";
+            query = "SELECT * FROM interval_test where timestamp IN '2022-11-21'";
+            assertSql(query, expected);
+        });
+    }
+
+    @Test
+    public void testTimestampIntervalPartitionMonth() throws Exception {
+        assertMemoryLeak(() -> {
+            //create table
+            String createStmt = "create table interval_test(seq_num long, timestamp timestamp) timestamp(timestamp) partition by MONTH";
+            compiler.compile(createStmt, sqlExecutionContext);
+            //insert as select
+            compiler.compile("insert into interval_test select x, timestamp_sequence(" +
+                    "'2022-11-19T00:00:00', " +
+                    Timestamps.DAY_MICROS * 30 + ") FROM long_sequence(5)", sqlExecutionContext);
+            String expected = "seq_num\ttimestamp\n" +
+                    "1\t2022-11-19T00:00:00.000000Z\n" +
+                    "2\t2022-12-19T00:00:00.000000Z\n" +
+                    "3\t2023-01-18T00:00:00.000000Z\n" +
+                    "4\t2023-02-17T00:00:00.000000Z\n" +
+                    "5\t2023-03-19T00:00:00.000000Z\n";
+            String query = "select * from interval_test";
+            assertSql(query, expected);
+            // test mid-case
+            expected = "seq_num\ttimestamp\n" +
+                    "3\t2023-01-18T00:00:00.000000Z\n";
+            query = "SELECT * FROM interval_test where timestamp IN '2023-01'";
+            assertSql(query, expected);
+        });
+    }
+
+    @Test
+    public void testTimestampIntervalPartitionWeek() throws Exception {
+        assertMemoryLeak(() -> {
+            //create table
+            String createStmt = "create table interval_test(seq_num long, timestamp timestamp) timestamp(timestamp) partition by WEEK";
+            compiler.compile(createStmt, sqlExecutionContext);
+            //insert as select
+            compiler.compile("insert into interval_test select x, timestamp_sequence(" +
+                    "'2022-11-19T00:00:00', " +
+                    Timestamps.WEEK_MICROS + ") FROM long_sequence(5)", sqlExecutionContext);
+            String expected = "seq_num\ttimestamp\n" +
+                    "1\t2022-11-19T00:00:00.000000Z\n" +
+                    "2\t2022-11-26T00:00:00.000000Z\n" +
+                    "3\t2022-12-03T00:00:00.000000Z\n" +
+                    "4\t2022-12-10T00:00:00.000000Z\n" +
+                    "5\t2022-12-17T00:00:00.000000Z\n";
+            String query = "select * from interval_test";
+            assertSql(query, expected);
+            // test mid-case
+            expected = "seq_num\ttimestamp\n" +
+                    "3\t2022-12-03T00:00:00.000000Z\n";
+            query = "SELECT * FROM interval_test where timestamp IN '2022-12-03'";
```

Comment:
We probably need to use ISO week format in the IN clause here, e.g. timestamp IN '2022W03'

Reply:
I did not implement ISO Weeks formatting in `IntervalUtils`. Is this something that should be added?

Reply:
Sorry, I missed that the parsing is not added in this PR. I'm afraid that it would be too much you to do that. We can address that ourselves, in a separate PR.